resources/sdk/pureclouddotnet/templates/ApiClient.mustache

```diff
@@ -156,6 +169,44 @@ namespace {{packageName}}.Client
             return request;
         }
 
+        private bool HasAccessTokenExpired(int statusCode)
+        {
+            TimeSpan timeSinceEpoch = (DateTime.UtcNow - new DateTime(1970, 1, 1));
+            long timeElapsed = (long) (timeSinceEpoch.TotalSeconds) - TokenIssueTime;
+            return statusCode == 401 && timeElapsed > TokenTimeToLive;
+        }
+
+        private void HandleExpiredAccessToken()
+        {
+            if (Monitor.TryEnter(RefreshTokenLock, 0))
+            {
+                try
+                {
+                    AuthTokenInfo authTokenInfo = Extensions.AuthExtensions.PostToken(this, ClientId, ClientSecret, authorizationCode: RefreshToken, isRefreshRequest: true);
+                    Configuration.Default.AccessToken = authTokenInfo.AccessToken;
```

Comment:
`Configuration.Default.AccessToken` is incorrect here. That's accessing the singleton default instance of Configuration, which may be different than the instance being used by this instance of ApiClient. I believe it should use `this.Configuration.AccessToken`.